src/SFA.DAS.Employer.PR.Web/Controllers/SelectTrainingProviderController.cs

```diff
@@ -12,28 +13,36 @@ namespace SFA.DAS.Employer.PR.Web.Controllers;
 
 [Authorize(Policy = nameof(PolicyNames.HasEmployerOwnerAccount))]
 [Route("accounts/{employerAccountId}/providers/new/selectProvider", Name = RouteNames.SelectTrainingProvider)]
-public class SelectTrainingProviderController(ISessionService _sessionService, IValidator<SelectTrainingProviderSubmitViewModel> _validator) : Controller
+public class SelectTrainingProviderController(IOuterApiClient _outerApiClient, ISessionService _sessionService, IValidator<SelectTrainingProviderSubmitModel> _validator) : Controller
 {
+    public const string ShutterPageViewPath = "~/Views/SetPermissions/ShutterPage.cshtml";
+
     [HttpGet]
     public IActionResult Index([FromRoute] string employerAccountId)
     {
         var sessionModel = _sessionService.Get<AddTrainingProvidersSessionModel>();
 
-        if (sessionModel == null)
+        if (sessionModel?.SelectedLegalEntityId == null || sessionModel.EmployerAccountId != employerAccountId)
```

Comment:
`employerAccountId` check is redundant, this should be the same as  in the url and without which you cannot be on this action. I dont think we need this in the session model either.

Reply:
as discussed, leaving this in